client/src/components/ReadingList/ReadingList.test.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { blankPaper } from '../../templates';
+import { getBlankInitialState, renderWithRouterRedux } from '../../testUtils/reduxRender';
+import { RootState } from '../../store';
+import { ReadingListState } from '../../slices/readingListSlice';
+import ReadingList from '.';
+
+describe('<ReadingList />', () => {
+  const readingList: ReadingListState = [];
+  const initialState: RootState = { ...getBlankInitialState(), readingList };
+
+  it('shows the option to add a manual entry', () => {
+    renderWithRouterRedux(<ReadingList />, { initialState });
+    expect(screen.getByText(/Add to Reading List/)).toBeInTheDocument();
+  });
+
+  describe('when the user clicks the manual add button', () => {
+    it('allows papers to be added', async () => {
+      renderWithRouterRedux(<ReadingList />, { initialState });
+      const startButton = screen.getByRole('button');
+      userEvent.click(startButton);
+
+      // wait for Input to appear
+      await screen.findByPlaceholderText('Paper title');
+
+      // type in a title
+      userEvent.type(screen.getByPlaceholderText('Paper title'), 'Everything about giraffes');
+
+      // click the accept button
+      const acceptButton = screen.getByRole('button', { name: 'submit' });
+      userEvent.click(acceptButton);
+
+      // see that our paper made it in
+      expect(screen.getByText(/giraffes/)).toBeInTheDocument();
+    });
+
+    it('allows you to bail out', async () => {
+      renderWithRouterRedux(<ReadingList />, { initialState });
+      const startButton = screen.getByRole('button');
+      userEvent.click(startButton);
+
+      // wait for Input to appear
+      await screen.findByPlaceholderText('Paper title');
+
+      // type in a title
+      userEvent.type(screen.getByPlaceholderText('Paper title'), 'Everything about giraffes');
+
+      // click the cancel button
+      const cancelButton = screen.getByRole('button', { name: 'cancel' });
+      userEvent.click(cancelButton);
+
+      // see that our paper did not make it in
+      expect(screen.queryByText(/giraffes/)).not.toBeInTheDocument();
+    });
```

Comment:
This block shares quite a bit with the previous one. @aradmargalit is there an obvious way to pull the common bits out into a common render, then "branch" the user logic in to a "submit" path and a "cancel" path?

Reply:
Sure! 
```ts
describe('when the user clicks the manual add button', () => {
  beforeEach(async () => {
    renderWithRouterRedux(<ReadingList />, { initialState });
    const startButton = screen.getByRole('button');
    userEvent.click(startButton);

    // wait for Input to appear
    await screen.findByPlaceholderText('Paper title');

    // type in a title
    userEvent.type(screen.getByPlaceholderText('Paper title'), 'Everything about giraffes');
  });
  
  it('allows papers to be added', async () => {
    // click the accept button
    const acceptButton = screen.getByRole('button', { name: 'submit' });
    userEvent.click(acceptButton);

    // see that our paper made it in
    expect(screen.getByText(/giraffes/)).toBeInTheDocument();
  });

  it('allows you to bail out', async () => {
    // click the cancel button
    const cancelButton = screen.getByRole('button', { name: 'cancel' });
    userEvent.click(cancelButton);

    // see that our paper did not make it in
    expect(screen.queryByText(/giraffes/)).not.toBeInTheDocument();
  });
});
```

Like this?